### ISSUE TITLE
Can't build slade with gcc

### ISSUE THREAD

**Issue by Monsterovich** [sirjuddington/SLADE#1051]
**SLADE Version**: master
**Operating System**: Linux Mint 19.1

**Issue Details**:
https://pastebin.com/SQpVevHp

```
monsterovich@dell ~/Distfiles/compile/SLADE/compile $ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.4.0-1ubuntu1~18.04' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1~18.04)
```

**Comment by Monsterovich** [comment]
Works successfully with clang. https://pastebin.com/18UU6Jgp

**Comment by axredneck** [comment]
Arch Linux, GCC 8.3.0, can't build too but with different error:
```In file included from /tmp/makepkg/slade-git/src/slade/src/Archive/ArchiveEntry.cpp:39:
/tmp/makepkg/slade-git/src/slade/src/./Utility/StringUtils.h:163:9: error: declaration of 'std::__cxx11::string StrUtil::Transformer::string' [-fpermissive]
  string string;
         ^~~~~~
In file included from /usr/include/c++/8.3.0/string:39,
                 from /usr/include/wx-3.0/wx/stringimpl.h:66,
                 from /usr/include/wx-3.0/wx/unichar.h:15,
                 from /usr/include/wx-3.0/wx/strvararg.h:22,
                 from /usr/include/wx-3.0/wx/string.h:46,
                 from /tmp/makepkg/slade-git/src/slade/src/common.h:5,
                 from /tmp/makepkg/slade-git/src/slade/src/cotire/slade_CXX_prefix.cxx:4,
                 from /tmp/makepkg/slade-git/src/slade/src/cotire/slade_CXX_prefix.hxx:4,
                 from <command-line>:
/usr/include/c++/8.3.0/bits/stringfwd.h:74:33: error: changes meaning of 'string' from 'typedef class std::__cxx11::basic_string<char> std::__cxx11::string' [-fpermissive]
   typedef basic_string<char>    string;

**Comment by sirjuddington** [comment]
The master branch uses C++17 features, so I think you need at least GCC 8 to compile. @axredneck's error should be fixed now too.
